### PR TITLE
feat(split-button): add SplitButton component with unit tests

### PR DESCRIPTION
**Pull request by dgolovin**
### What does this PR do?

- Introduces a new SplitButton component in Svelte, allowing for single and multi-selection options.
- Implemets various features including dropdown functionality, option selection, and action handling.
- Provides fine tuning for corner cases: button name when no option provided, button name when nothing selected from options selected 
- Component unit tests

### Screenshot / video of UI

https://github.com/user-attachments/assets/b39097f2-ca49-4edc-a6f8-e4906cc3c7ed

### What issues does this PR fix or reference?

Related to #14729.

### How to test this PR?

- [x] Tests are covering the new UI control functionality

### CODE REVIEW

File: packages/ui/src/lib/button/SplitButton.spec.ts [hunk 1]
```diff
@@ -0,0 +1,658 @@
+/**********************************************************************
+ * Copyright (C) 2024-2025 Red Hat, Inc.
```
Comment (dgolovin) [inline packages/ui/src/lib/button/SplitButton.spec.ts]:
```suggestion
 * Copyright (C) 2026 Red Hat, Inc.
```

**Comment by simonrey1** [comment]
suggestion: add export of the new SplitButton component to `packages/ui/src/lib/index.ts`

**Review by simonrey1 — COMMENTED** [review]
Thanks for creating this new UI component.
I think the classes could be given explicitly in a `Record`, rather than having the function `getButtonClasses`. Here is a suggestion PR to do so: https://github.com/dgolovin/podman-desktop/pull/1/changes. The PR also adds the SplitButton to the exports of the UI components, that was helpful for testing and will be needed to use this new component.

**Comment by MariaLeonova** [comment]
@dgolovin hi! This looks great! I love the split button for more options. 
What happens if the user clicks on the button while no machines are selected yet? 
I also wonder if there's a way to keep button label more consistent, so it doesn't jump around quite so much.

That could potentially be achieved by it always saying `Synchronize to...` or `Synchronize to #` 1, 2 or 3, depending on the state. 
When nothing is available, like in the end of the video, it can be simply disabled. That should be clear enough indication.

**Comment by dgolovin** [comment]
@MariaLeonova 

> What happens if the user clicks on the button while no machines are selected yet?

It would show the list of the options.

> I also wonder if there's a way to keep button label more consistent, so it doesn't jump around quite so much. That could potentially be achieved by it always saying `Synchronize to...` or `Synchronize to #` 1, 2 or 3, depending on the state. When nothing is available, like in the end of the video, it can be simply disabled. That should be clear enough indication.

That should be done on Certificates Preference page.

**Comment by dgolovin** [comment]
> Thanks for creating this new UI component. I think the classes could be given explicitly in a `Record`, rather than having the function `getButtonClasses`. Here is a suggestion PR to do so: https://github.com/dgolovin/podman-desktop/pull/1/changes. The PR also adds the SplitButton to the exports of the UI components, that was helpful for testing and will be needed to use this new component.

@simonrey1 thank you for the PR.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+// Check if no option is selected (for button display purposes)
+const hasNoSelection = $derived(!isEmpty && hasActionableOption && selectedOptionIds.length === 0);
+// Main action is disabled if no valid selection or selected option is disabled
+const isMainDisabled = $derived(disabled || isEmpty || (selectedOption?.disabled ?? false));
+// Dropdown can be opened even if selected option is disabled (to see/change options)
+const isDropdownDisabled = $derived(disabled || isEmpty);
+// Show "no action" mode when options exist but none are actionable
+const isNoActionMode = $derived(!isEmpty && !hasActionableOption);
+// Show "no selection" mode when options exist and are actionable but none selected
+const isNoSelectionMode = $derived(hasNoSelection);
```
Comment (simonrey1) [inline packages/ui/src/lib/button/SplitButton.svelte]: suggestion: you can rmeove this intermediate const, and use directly `hasNoSelection`

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 3]
```diff
@@ -0,0 +1,297 @@
+<script lang="ts">
+import type { IconDefinition } from '@fortawesome/free-solid-svg-icons';
+import { faCheck, faChevronDown } from '@fortawesome/free-solid-svg-icons';
+import type { Component, Snippet } from 'svelte';
+
+import Icon from '../icons/Icon.svelte';
+import Spinner from '../progress/Spinner.svelte';
```
Comment (benoitf) [inline packages/ui/src/lib/button/SplitButton.svelte:7]: issue: should not use relative imports

Comment (dgolovin) [reply packages/ui/src/lib/button/SplitButton.svelte:7]: @benoitf  Everywhere in `ui` package relative path is used because `/@/` alias cannot be used for the following reasons:
1. svelte-ui package using svelte-package and that creates copies of .svelte files (with `/@/` aliases in them)
2. renderer package resolves referenced components from svelte-ui to packages/ui/dist/indext.ts
3. renderer vite config also has `/@/` alias definition which resolves to packages/renderer/src and that one is used to resolve aliases for svelte-ui files from packages/ui/dist/.
4. renderer build fails to resolve it with error `Rollup failed to resolve import "/@/icons/Icon.svelte" from "/Users/eskimo/Sources/podman-desktop/packages/ui/dist/button/SplitButton.svelte". ` because `/@/` resolves to `packages/renderer/src`

The solution would be to use @svelte-ui (or `@ui`) alias in `ui` package that would point to the same location as `/@/` alias, then also define @svelte-ui alias for `renderer` package that would resolve to `packages/ui/dist`.

Then we can migrate other 24 files with ../ imports and add linting rule to disallow relative imports.

Comment (dgolovin) [reply packages/ui/src/lib/button/SplitButton.svelte:7]: Fix for the /@/ alias conflict https://github.com/podman-desktop/podman-desktop/pull/16203

**Comment by simonrey1** [comment]
As I co-authored this PR https://github.com/podman-desktop/podman-desktop/pull/16033/changes/750817728abb5e533d368caf17fd94137bf9ac2b should I approve it?

File: packages/ui/src/lib/button/SplitButton.svelte (hunk 3 above)
Comment (dgolovin) [reply packages/ui/src/lib/button/SplitButton.svelte:7]: Reverted back to relative path. Checked other options and it seems $lib internal alias works, but let's do it later.

**Comment by dgolovin** [comment]
> As I co-authored this PR [7508177](https://github.com/podman-desktop/podman-desktop/commit/750817728abb5e533d368caf17fd94137bf9ac2b) should I approve it?

@simonrey1 I would say it does not matter who implemented requested changes. IMO you can approve it. But I would still prefer to see one more approval for new UI element before releasing it into main.

**Comment by MariaLeonova** [comment]
@dgolovin can you please update the video in the description for review?

**Review by simonrey1 — APPROVED** [review]
LGTM but disclaimer I co-authored this PR

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+        'border-2 border-[var(--pd-button-danger-border)] bg-[var(--pd-button-danger-bg)] text-[var(--pd-button-danger-text)] hover:bg-[var(--pd-button-danger-hover-bg)] hover:text-[var(--pd-button-danger-hover-text)]',
+      disabled:
+        'border-2 border-[var(--pd-button-danger-disabled-border)] text-[var(--pd-button-danger-disabled-text)] bg-[var(--pd-button-danger-disabled-bg)]',
+    },
+    divider: 'bg-[var(--pd-button-danger-text)]',
+    border: 'border-[var(--pd-button-danger-border)]',
+    containerBg: 'bg-[var(--pd-button-danger-bg)]',
+    focusOutline: 'focus:outline-[var(--pd-button-danger-hover-bg)]',
+  },
+};
```
Comment (cdrage) [inline packages/ui/src/lib/button/SplitButton.svelte:264]: I feel like the above would add maintenance burden when updating styles for the buttons, but digging through some other Svelte files.. I don't have an alternative other than pushing some changes to Button.ts 

(non-blocker / nitpick)

**Review by cdrage — APPROVED** [review]
Looks great! 

Tested it out too and it was cool seeing it dropdown.

Nitpick, but should we be adding arrow down / arrow up / enter support like Dropdown.svelte?

**Comment by dgolovin** [comment]
> @dgolovin can you please update the video in the description for review?

@MariaLeonova attached updated split button video.

**Review by MariaLeonova — APPROVED** [review]
@dgolovin looks great! Thank you for making these updates!

**Review by vancura — CHANGES_REQUESTED** [review]
I haven't seen this before, and I wasn't involved in the approvals. I need to test this thoroughly before you merge it. I already see an issue.

**Comment by vancura** [comment]
PLEASE remember to add me as a reviewer when the design system or UI components are involved. Merging without my knowledge should NOT happen.

**Comment by benoitf** [comment]
@dgolovin what is the state of this PR ? should it be fixed, closed ?

**Comment by vancura** [comment]
Hey, I would like to review, but we need #16966 to be merged first. Then I'll be able to see what needs to be done here visually. Please keep this open until I am able to review. Thanks!

**Comment by vancura** [comment]
#16966 (new buttons) is now merged to main, so I will update the branch with these changes. I hope it's fine – I will review then.

**Comment by vancura** [comment]
@dgolovin: Where can I test the `SplitButton` please? It seems it's not used anywhere, and there are no Storybook stories for it.

**Comment by vancura** [comment]
I vibecoded a couple of `SplitButton` Storybook stories, and here's what I see:

1. It's possible to have secondary and danger `SplitButton`s, yet only the primary button is visually supported:

CleanShot 2026-04-10 at 14 46 49@2x

2. Our new button has a lighter stroke (especially visible in the dark theme), yet the `SplitButton` doesn't render it:

`SplitButton`:

CleanShot 2026-04-10 at 14 48 19@2x

`Button` (see the stroke):

CleanShot 2026-04-10 at 14 48 55@2x

4. The stroke is also missing from high contrast themes:

`SplitButton`:

CleanShot 2026-04-10 at 14 49 12@2x

`Button` (see the stroke):

CleanShot 2026-04-10 at 14 49 35@2x

These are my main visual issues. Could you please take a look at them? Would you like me to push the Storybook story somewhere, even though it's quite temporary? We should also write a proper story that includes everything necessary.

**Review by benoitf — APPROVED** [review]
approving from code pov but still require the approval from UX

**Comment by dgolovin** [comment]
@vancura could you create a branch with the storybook and push it to your fork? I'll take it from there.

**Comment by vancura** [comment]
@dgolovin https://github.com/podman-desktop/podman-desktop/pull/17041

**Review by vancura — CHANGES_REQUESTED** [review]
I can't approve it due to many issues, as seen in the stories I created (https://github.com/podman-desktop/podman-desktop/pull/17041).

**Comment by vancura** [comment]
@dgolovin Please use the stories from https://github.com/podman-desktop/podman-desktop/pull/17041 to fix the component. Once the problems are fixed, please add the stories to your PR. All new components must have Storybook stories.

**Comment by dgolovin** [comment]
@vancura I don't think the SplitButton component can be used in as suggested [here ](https://github.com/podman-desktop/podman-desktop/issues/16370#issuecomment-4178239326)

But it can be used the same way it is used on github.com to select Merge or Rebase strategy to release PR changes to targeted branch. Below the prototype with SplitButton in current implementation from this PR. It would eliminate additional modal dialog and save selected way to create container.

If it is still relevant, I can work on fixing design later.

https://github.com/user-attachments/assets/8adb5b64-c3e7-43a8-b02a-45501a985db5

**Comment by vancura** [comment]
Yes, I believe `SplitButton` **is** a helpful component, and we can use it in some places. Let's work on it together later, when the time comes. I hope [my Storybook branch](https://github.com/podman-desktop/podman-desktop/pull/17041) was helpful to you. Let's keep it open for when we're here again, ok?

**Comment by cdrage** [comment]
> Yes, I believe `SplitButton` **is** a helpful component, and we can use it in some places. Let's work on it together later, when the time comes. I hope [my Storybook branch](https://github.com/podman-desktop/podman-desktop/pull/17041) was helpful to you. Let's keep it open for when we're here again, ok?

👍 on this being useful. I would love this added so I can use it in the bootc extension.

We also had a mockup by Masha to improve the container form workflow, and one of the suggestions was to use a drop-down design rather than a modal to select "Create from Containerfile vs Use existing image".

So big 👍 in usefulness for this component being added.

**Comment by cdrage** [comment]
We also have 4 approvals in this PR

**Comment by dgolovin** [comment]
> We also have 4 approvals in this PR

@cdrage design issues are holding it from being merged. I'll find time to work on those to get approval from @vancura.

**Comment by cdrage** [comment]
@vancura Could we have a review of this / ready to go?

It'd be nice to get into https://github.com/podman-desktop/podman-desktop/issues/16370 which requires the split button :)

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+  secondary: {
+    button: {
+      enabled:
+        'border-[1px] border-[var(--pd-button-secondary)] text-[var(--pd-button-secondary)] hover:bg-[var(--pd-button-secondary-hover)] hover:border-[var(--pd-button-secondary-hover)] hover:text-[var(--pd-button-text)]',
+      disabled:
+        'border-[1px] border-[var(--pd-button-disabled)] bg-[var(--pd-button-disabled)] text-[var(--pd-button-disabled-text)]',
+    },
+    divider: 'bg-[var(--pd-button-secondary)]',
+    border: 'border-[var(--pd-button-secondary)]',
+    containerBg: 'bg-[var(--pd-button-disabled)]',
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte]: `--pd-button-disabled` is set as the disabled background, even when the button is fully active. This makes every non-disabled secondary `SplitButton` render with a grayed-out container. The correct token here should match `secondary.border` (`--pd-button-secondary`) or be transparent.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+// Dropdown can be opened even if selected option is disabled (to see/change options)
+const isDropdownDisabled = $derived(disabled || isEmpty);
+// Show "no action" mode when options exist but none are actionable
+const isNoActionMode = $derived(!isEmpty && !hasActionableOption);
+
+function handleEscape({ key }: KeyboardEvent): void {
+  if (key === 'Escape') {
+    showDropdown = false;
+  }
+}
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte:92]: The dropdown handles only `Escape`. The ARIA Authoring Practices Guide listbox pattern requires `ArrowUp`/`Down` to navigate options, `Enter` to select the focused item, and `Home`/`End` to jump to first/last.

**Without this, keyboard-only users can open the dropdown but cannot operate it.**

The existing `Dropdown.svelte` in this repo implements the full pattern and can serve as a reference.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+            <Icon icon={option.icon} class="w-4" />
+          {/if}
+          <div class="flex flex-col">
+            <span class="font-medium">{option.label}</span>
+            {#if option.description}
+              <span class="text-xs text-[var(--pd-dropdown-item-description-text)] opacity-70"
+                >{option.description}</span>
+            {/if}
+          </div>
+        </button>
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte]: Each option is a `<button>` with `role="option"` inside a `div[role="listbox"]`. The ARIA spec does not allow button as a child of `listbox`. The `option` role must be on a non-interactive element.

**Screen readers will announce "button" instead of "option" and the widget will fail accessibility tree validation.**

Fix: replace the `<button>` with a `<div role="option" tabindex="0">` handling click and keydown (`Enter`/`Space`).

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+      <Icon icon={faChevronDown} class="w-3 text-current" />
+    </button>
+  </div>
+
+  <!-- Dropdown menu -->
+  {#if showDropdown}
+    <div
+      class="absolute right-0 mt-1 min-w-full w-max bg-[var(--pd-dropdown-bg)] border border-[var(--pd-dropdown-border)] rounded-md shadow-lg z-50"
+      role="listbox"
+      aria-multiselectable={multipleSelection}>
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte:354]: The `div[role="listbox"]` has no `aria-label` or `aria-labelledby`.

**Screen readers will not announce what the listbox represents when it receives focus.**

Fix: add `aria-label="Select action"` (or wire it from a prop so callers can customize it).

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+      aria-haspopup="listbox"
+      aria-expanded={showDropdown}
+      onclick={handleDropdownClick}
+      disabled={isDropdownDisabled || inProgress}>
+      <Icon icon={faChevronDown} class="w-3 text-current" />
+    </button>
+  </div>
+
+  <!-- Dropdown menu -->
+  {#if showDropdown}
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte:349]: When the dropdown appears, focus stays on the toggle button. The APG split button pattern requires focus to move to the first option (or the listbox itself) on open.

On `Escape` or selection, focus must explicitly return to the toggle button. 

There is no `focus()` call or `use: action` handling this.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+const buttonLabel = $derived.by(() => {
+  if (isEmpty) return emptyLabel;
+  if (isNoActionMode) return noActionLabel;
+  if (hasNoSelection) return noSelectionLabel;
+  if (!children && selectedOption) return selectedOption.label;
+  return undefined;
+});
+</script>
+
+<svelte:window onkeyup={handleEscape} onclick={onWindowClick} />
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte:302]: `onkeyup={handleEscape}` fires after the key is released. The standard (and what `Dropdown.svelte` uses) is `onkeydown`, which gives immediate response and allows `preventDefault()` to suppress default browser behavior.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+  primary: {
+    button: {
+      enabled:
+        'bg-[var(--pd-button-primary-bg)] text-[var(--pd-button-text)] hover:bg-[var(--pd-button-primary-hover-bg)]',
+      disabled: 'bg-[var(--pd-button-disabled)] text-[var(--pd-button-disabled-text)]',
+    },
+    divider: 'bg-[var(--pd-button-text)]',
+    border: 'border-[var(--pd-button-primary-bg)]',
+    containerBg: 'bg-[var(--pd-button-primary-bg)]',
+    focusOutline: 'focus:outline-[var(--pd-button-primary-hover-bg)]',
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte]: All three button types apply `focus:outline-[...]`. This shows the focus ring on mouse click as well as keyboard focus.

`focus-visible:outline-[...]` is the modern pattern used in the rest of the codebase and limits the ring to keyboard navigation.

Also :162 and :174.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+
+<svelte:window onkeyup={handleEscape} onclick={onWindowClick} />
+
+<div class="relative inline-block" bind:this={buttonRef}>
+  <!-- Split button container -->
+  <div class="flex flex-row items-center rounded-[4px] overflow-hidden border {borderClasses} {containerBgClasses} {classNames}">
+    <!-- Main action button -->
+    <button
+      type="button"
+      class="relative px-4 py-[5px] whitespace-nowrap select-none transition-all outline-transparent {currentStyle.focusOutline} rounded-l-[4px] {mainClasses}"
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte]: Both action button and toggle button use `transition-all` without `motion-reduce:transition-none`.

Also :248.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+          onclick={(): void => selectOption(option)}>
+          <!-- Checkmark for selected option(s) -->
+          <span class="w-4 flex-shrink-0">
+            {#if selected && !option.disabled}
+              <Icon icon={faCheck} class="w-4 text-current" />
+            {/if}
+          </span>
+          {#if option.icon}
+            <Icon icon={option.icon} class="w-4" />
+          {/if}
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte:380]: `<Icon icon={option.icon} />` inside option buttons is decorative (the text label describes the option). It should carry `aria-hidden="true"` so screen readers do not announce it separately.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+          type="button"
+          class="w-full text-left px-4 py-2 first:rounded-t-md last:rounded-b-md flex items-center gap-2
+            {option.disabled
+              ? 'text-[var(--pd-dropdown-item-disabled-text)] cursor-not-allowed opacity-60'
+              : 'hover:bg-[var(--pd-dropdown-item-hover-bg)] hover:text-[var(--pd-dropdown-item-hover-text)] text-[var(--pd-dropdown-item-text)]'}"
+          class:bg-[var(--pd-dropdown-item-selected-bg)]={selected && !option.disabled}
+          role="option"
+          aria-selected={selected}
+          aria-disabled={option.disabled}
+          disabled={option.disabled}
```
Comment (vancura) [inline packages/ui/src/lib/button/SplitButton.svelte]: Options carry both the native `disabled` attribute and `aria-disabled`.

Native `disabled` removes the element from focus order entirely, making `aria-disabled` unreachable by keyboard.

For `role="option"` items (once fixed, see other nitpicks), use only `aria-disabled` and guard `selectOption()` in the click handler.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 15]
```diff
@@ -0,0 +1,297 @@
...
+          <div class="flex flex-col">
+            <span class="font-medium">{option.label}</span>
+            {#if option.description}
+              <span class="text-xs text-[var(--pd-dropdown-item-description-text)] opacity-70"
+                >{option.description}</span>
+            {/if}
+          </div>
+        </button>
+      {/each}
+    </div>
```
[the remark this answers is not in the document]
Comment (vancura) [reply packages/ui/src/lib/button/SplitButton.svelte:390]: @dgolovin Also this please.

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 16]
```diff
@@ -0,0 +1,295 @@
...
+          role="option"
+          aria-selected={selected}
+          aria-disabled={option.disabled}
+          disabled={option.disabled}
+          onclick={(): void => selectOption(option)}>
+          <!-- Checkmark for selected option(s) -->
+          <span class="w-4 flex-shrink-0">
+            {#if selected && !option.disabled}
+              <Icon icon={faCheck} class="w-4 text-current" />
+            {/if}
```
[the remark this answers is not in the document]
Comment (vancura) [reply packages/ui/src/lib/button/SplitButton.svelte:376]: Don't miss this one either, @dgolovin.

**Review by vancura — CHANGES_REQUESTED** [review]
Sorry, I can't approve these changes. The issues I pointed at in https://github.com/podman-desktop/podman-desktop/pull/16033#issuecomment-4223866291 are still unresolved. The Storybook stories are untouched. Here are current screenshots, but you can test them in #17041.

Light:

PixelSnap 2026-06-17 at 10 01 31@2x

Dark:

PixelSnap 2026-06-17 at 10 04 24@2x

HC Light:

PixelSnap 2026-06-17 at 10 04 57@2x

HC Dark:

PixelSnap 2026-06-17 at 10 05 26@2x

Secondary and danger types don't work at all. If you don't want to support them, disable the prop. Their rollovers are even worse, but if we don't support these types, I won't report them here. If we do, there's further work to do on them too.

FYI, there are many Tailwind 3 bracket forms. We should use Tailwind v4:

- `bg-[var(--pd-button-primary-bg)]` → `bg-(--pd-button-primary-bg`
- `text-[var(--pd-button-text)]` → `text-(--pd-button-text)`
- `border-[var(--pd-button-secondary)]` → `border-(--pd-button-secondary)`
- `hover:bg-[var(--pd-dropdown-item-hover-bg)]` → `hover:bg-(--pd-dropdown-item-hover-bg)`

See my other nitpicks below.

**Comment by dgolovin** [comment]
> @vancura Could we have a review of this / ready to go?
> 
> It'd be nice to get into #16370 which requires the split button :)

It might as well fit for https://github.com/podman-desktop/podman-desktop/issues/17752 where configure button would show drop down and allow to switch between username/password, SSO or any other option.

File: packages/ui/src/lib/button/SplitButton.svelte (hunk 13 above)
Comment (dgolovin) [reply packages/ui/src/lib/button/SplitButton.svelte:380]: Icon doesn't accept an aria-hidden prop and doesn't spread $$restProps / ...rest. The role="img" is hardcoded too, so there's no current way to pass through aria-hidden="true" without changing the component.

The cleanest fix is to add an optional ariaHidden prop to Icon component as fix for this specific issue.

**Comment by benoitf** [comment]
closing as outdated
